File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        max_num = 0
        for i in range(k):
            max_num += nums[i]    

        cur_num = max_num
        for j in range(k, len(nums)):
            cur_num -= nums[j-k]
            cur_num += nums[j]
            max_num = max(max_num, cur_num)

        return max_num / k

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        // maxの初期値を計算
        int max = 0;
        for(int i = 0; i < k; i++) {
            max += nums[i];
        }

        int cur_total = max;
        for(int j = k; j < nums.length; j++) {
            cur_total += nums[j];
            cur_total -= nums[j-k];
            if (cur_total > max) {
                max = cur_total;
            }
        }

        double max_f = max;
        return max_f / k;
    }
}